### DISCUSSION TITLE
IFC to IFCxml format for visualization

### DISCUSSION THREAD

**Discussion by Thirumurugan-12** [IfcOpenShell/IfcOpenShell#4315] [Q&A] [answered]
I've utilized IFCconvert to convert an IFC file to XML format, but unfortunately, it doesn't produce the desired IFCXML output. Can anyone advise on how to troubleshoot this issue or suggest alternative methods/tools to generate the required IFCXML format? Thank you!

```
<IFCWall > id="3M_OrpkUn8Oxs41366u50e" ObjectPlacement=-0.019607 0.999808 0 0-0.999808 0.019607 0 0 0 0 1 0 173076 I 63815 -23540 1" Tag="D6F98073-89EC-4803-13084-A43186E3F628"> 
  < IfcPropertySet xlinlehref ="01x4b2DkUOYJI1Lhystrz3SA"/> 
  <I1cPropertySet xlinkihref ="*OltXttlF8moTrGvEiczLyLhr-/> 
  <IfcWallType xlinthref="41i2P6ZoQx72PD1WzkX_kgEls"/> 
  <IfcPresentationLayerAssIgnment xlinIchref ="*INV123 RUNK0.1N■0> 
</IfcWall> 
```

Now can we somehow make this like the following style : 

```
<Wall id." (9c1c68d47 -be9b-445a-8b32-72bct3ca 18e85r > 
<StartPoint>14.115509.21.890991</StartPoint> 
<EndPoint> -29.721600.23.915520</EndPoint> 
<Height >16390000 </Height > 
<Thickness>0.100000</Thickness> 
<Material > 0</Matenal> 
<CustomColor > 1.000000.1.000000.1.000000 </CustomColor > 
<StartAttachedTo></StartAttachedTo> 
<EndAttacheciTo> </EndAttachedTo> 
<Entrances/> 
<C1C)Ps/> 
<CallPanels/> 
<Turns-Weser> 
<Windows/> 
<AutoGenerated >0 </AutoGenerated </Wall> 
```

**Comment by Moult** [comment]
That's not how IFCXML works. IFCXML is an official serialisation of the IFC schema (just like IFC-SPF, or the ".ifc" format). There are rules on what it should contain and how. What you're after is a custom set of data.

The good news is that it's pretty easily extracted with Python. Would you be able to write some code?

**Reply by Thirumurugan-12** [reply]
Can I know how it can be extracted using python script??

**Reply by aothms** [reply, marked as answer]
This information is most likely not directly explicitly contained in the model so needs to be calculated. There's some resources available on how to do that. https://academy.ifcopenshell.org/ https://community.osarch.org But it will likely require some dedication on your end to properly specify the information you need, how to map it to IFC concepts and to do the implementation.

**Reply by Thirumurugan-12** [reply]
Thank you
